--- a/com/denizenscript/denizen/objects/properties/PropertyRegistry.java
+++ b/com/denizenscript/denizen/objects/properties/PropertyRegistry.java
@@ -44,17 +_,13 @@
         PropertyParser.registerProperty(EntityArms.class, EntityTag.class);
         PropertyParser.registerProperty(EntityAwake.class, EntityTag.class);
         PropertyParser.registerProperty(EntityAware.class, EntityTag.class);
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
-            PropertyParser.registerProperty(EntityBackgroundColor.class, EntityTag.class);
-        }
+        PropertyParser.registerProperty(EntityBackgroundColor.class, EntityTag.class);
         PropertyParser.registerProperty(EntityBasePlate.class, EntityTag.class);
         PropertyParser.registerProperty(EntityBeamTarget.class, EntityTag.class);
         PropertyParser.registerProperty(EntityBoatType.class, EntityTag.class);
         PropertyParser.registerProperty(EntityBodyArrows.class, EntityTag.class);
         PropertyParser.registerProperty(EntityBoundingBox.class, EntityTag.class);
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
-            PropertyParser.registerProperty(EntityBrightness.class, EntityTag.class);
-        }
+        PropertyParser.registerProperty(EntityBrightness.class, EntityTag.class);
         PropertyParser.registerProperty(EntityCanBreakDoors.class, EntityTag.class);
         PropertyParser.registerProperty(EntityCanJoinRaid.class, EntityTag.class);
         PropertyParser.registerProperty(EntityCannotEnterHive.class, EntityTag.class);
@@ -67,20 +_,14 @@
         PropertyParser.registerProperty(EntityCustomName.class, EntityTag.class);
         PropertyParser.registerProperty(EntityCustomNameVisible.class, EntityTag.class);
         PropertyParser.registerProperty(EntityDarkDuration.class, EntityTag.class);
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
-            PropertyParser.registerProperty(EntityDefaultBackground.class, EntityTag.class);
-        }
+        PropertyParser.registerProperty(EntityDefaultBackground.class, EntityTag.class);
         PropertyParser.registerProperty(EntityDirection.class, EntityTag.class);
         PropertyParser.registerProperty(EntityDisabledSlots.class, EntityTag.class);
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
-            PropertyParser.registerProperty(EntityDisplay.class, EntityTag.class);
-        }
+        PropertyParser.registerProperty(EntityDisplay.class, EntityTag.class);
         PropertyParser.registerProperty(EntityDropsItem.class, EntityTag.class);
         PropertyParser.registerProperty(EntityEquipment.class, EntityTag.class);
         PropertyParser.registerProperty(EntityEquipmentDropChance.class, EntityTag.class);
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
-            PropertyParser.registerProperty(EntityExploredLocations.class, EntityTag.class);
-        }
+        PropertyParser.registerProperty(EntityExploredLocations.class, EntityTag.class);
         PropertyParser.registerProperty(EntityExplosionFire.class, EntityTag.class);
         PropertyParser.registerProperty(EntityExplosionRadius.class, EntityTag.class);
         PropertyParser.registerProperty(EntityEyeTargetLocation.class, EntityTag.class);
@@ -91,23 +_,17 @@
         PropertyParser.registerProperty(EntityFlower.class, EntityTag.class);
         PropertyParser.registerProperty(EntityFreezeDuration.class, EntityTag.class);
         PropertyParser.registerProperty(EntityFramed.class, EntityTag.class);
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
-            PropertyParser.registerProperty(EntityGlowColor.class, EntityTag.class);
-        }
+        PropertyParser.registerProperty(EntityGlowColor.class, EntityTag.class);
         PropertyParser.registerProperty(EntityGravity.class, EntityTag.class);
         PropertyParser.registerProperty(EntityHasNectar.class, EntityTag.class);
         PropertyParser.registerProperty(EntityHasStung.class, EntityTag.class);
         PropertyParser.registerProperty(EntityHealth.class, EntityTag.class);
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
-            PropertyParser.registerProperty(EntityHeight.class, EntityTag.class);
-            PropertyParser.registerProperty(EntityHorns.class, EntityTag.class);
-        }
+        PropertyParser.registerProperty(EntityHeight.class, EntityTag.class);
+        PropertyParser.registerProperty(EntityHorns.class, EntityTag.class);
         PropertyParser.registerProperty(EntityHive.class, EntityTag.class);
         PropertyParser.registerProperty(EntityImmune.class, EntityTag.class);
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
-            PropertyParser.registerProperty(EntityInterpolationDuration.class, EntityTag.class);
-            PropertyParser.registerProperty(EntityInterpolationStart.class, EntityTag.class);
-        }
+        PropertyParser.registerProperty(EntityInterpolationDuration.class, EntityTag.class);
+        PropertyParser.registerProperty(EntityInterpolationStart.class, EntityTag.class);
         PropertyParser.registerProperty(EntityInventory.class, EntityTag.class);
         PropertyParser.registerProperty(EntityInvulnerable.class, EntityTag.class);
         PropertyParser.registerProperty(EntityInWaterTime.class, EntityTag.class);
@@ -117,49 +_,35 @@
         PropertyParser.registerProperty(EntityItemInOffHand.class, EntityTag.class);
         PropertyParser.registerProperty(EntityJumpStrength.class, EntityTag.class);
         PropertyParser.registerProperty(EntityKnockback.class, EntityTag.class);
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
-            PropertyParser.registerProperty(EntityLeftRotation.class, EntityTag.class);
-            PropertyParser.registerProperty(EntityLineWidth.class, EntityTag.class);
-        }
+        PropertyParser.registerProperty(EntityLeftRotation.class, EntityTag.class);
+        PropertyParser.registerProperty(EntityLineWidth.class, EntityTag.class);
         PropertyParser.registerProperty(EntityMarker.class, EntityTag.class);
         PropertyParser.registerProperty(EntityMaterial.class, EntityTag.class);
         PropertyParser.registerProperty(EntityMaxFuseTicks.class, EntityTag.class);
         PropertyParser.registerProperty(EntityMaxTemper.class, EntityTag.class);
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
-            PropertyParser.registerProperty(EntityOnBack.class, EntityTag.class);
-            PropertyParser.registerProperty(EntityOpacity.class, EntityTag.class);
-        }
+        PropertyParser.registerProperty(EntityOnBack.class, EntityTag.class);
+        PropertyParser.registerProperty(EntityOpacity.class, EntityTag.class);
         PropertyParser.registerProperty(EntityPainting.class, EntityTag.class);
         PropertyParser.registerProperty(EntityPatrolLeader.class, EntityTag.class);
         PropertyParser.registerProperty(EntityPatrolTarget.class, EntityTag.class);
         PropertyParser.registerProperty(EntityPickupStatus.class, EntityTag.class);
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
-            PropertyParser.registerProperty(EntityPivot.class, EntityTag.class);
-        }
+        PropertyParser.registerProperty(EntityPivot.class, EntityTag.class);
         PropertyParser.registerProperty(EntityPlayerCreated.class, EntityTag.class);
         PropertyParser.registerProperty(EntityPlayingDead.class, EntityTag.class);
         PropertyParser.registerProperty(EntityPotion.class, EntityTag.class);
         PropertyParser.registerProperty(EntityPotionEffects.class, EntityTag.class);
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_20)) {
-            PropertyParser.registerProperty(EntityPotionType.class, EntityTag.class);
-        }
+        PropertyParser.registerProperty(EntityPotionType.class, EntityTag.class);
         PropertyParser.registerProperty(EntityPowered.class, EntityTag.class);
         PropertyParser.registerProperty(EntityProfession.class, EntityTag.class);
         PropertyParser.registerProperty(EntityRiptide.class, EntityTag.class);
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
-            PropertyParser.registerProperty(EntityRightRotation.class, EntityTag.class);
-            PropertyParser.registerProperty(EntityRolling.class, EntityTag.class);
-        }
+        PropertyParser.registerProperty(EntityRightRotation.class, EntityTag.class);
+        PropertyParser.registerProperty(EntityRolling.class, EntityTag.class);
         PropertyParser.registerProperty(EntityRotation.class, EntityTag.class);
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
-            PropertyParser.registerProperty(EntityScale.class, EntityTag.class);
-        }
+        PropertyParser.registerProperty(EntityScale.class, EntityTag.class);
         PropertyParser.registerProperty(EntityScoreboardTags.class, EntityTag.class);
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
-            PropertyParser.registerProperty(EntitySeeThrough.class, EntityTag.class);
-            PropertyParser.registerProperty(EntityShadowRadius.class, EntityTag.class);
-            PropertyParser.registerProperty(EntityShadowStrength.class, EntityTag.class);
-        }
+        PropertyParser.registerProperty(EntitySeeThrough.class, EntityTag.class);
+        PropertyParser.registerProperty(EntityShadowRadius.class, EntityTag.class);
+        PropertyParser.registerProperty(EntityShadowStrength.class, EntityTag.class);
         PropertyParser.registerProperty(EntitySheared.class, EntityTag.class);
         PropertyParser.registerProperty(EntityShivering.class, EntityTag.class);
         PropertyParser.registerProperty(EntityShotAtAngle.class, EntityTag.class);
@@ -168,48 +_,29 @@
         PropertyParser.registerProperty(EntitySitting.class, EntityTag.class);
         PropertyParser.registerProperty(EntitySize.class, EntityTag.class);
         PropertyParser.registerProperty(EntitySmall.class, EntityTag.class);
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
-            PropertyParser.registerProperty(EntitySneezing.class, EntityTag.class);
-        }
+        PropertyParser.registerProperty(EntitySneezing.class, EntityTag.class);
         PropertyParser.registerProperty(EntitySpeed.class, EntityTag.class);
         PropertyParser.registerProperty(EntitySpell.class, EntityTag.class);
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_20)) {
-            PropertyParser.registerProperty(EntityState.class, EntityTag.class);
-        }
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
-            PropertyParser.registerProperty(EntityStepHeight.class, EntityTag.class);
-        }
+        PropertyParser.registerProperty(EntityState.class, EntityTag.class);
+        PropertyParser.registerProperty(EntityStepHeight.class, EntityTag.class);
         PropertyParser.registerProperty(EntityStrength.class, EntityTag.class);
         PropertyParser.registerProperty(EntityTame.class, EntityTag.class);
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_20)) {
-            PropertyParser.registerProperty(EntityTeleportDuration.class, EntityTag.class);
-        }
+        PropertyParser.registerProperty(EntityTeleportDuration.class, EntityTag.class);
         PropertyParser.registerProperty(EntityTemper.class, EntityTag.class);
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
-            PropertyParser.registerProperty(EntityText.class, EntityTag.class);
-            PropertyParser.registerProperty(EntityTextShadowed.class, EntityTag.class);
-        }
+        PropertyParser.registerProperty(EntityText.class, EntityTag.class);
+        PropertyParser.registerProperty(EntityTextShadowed.class, EntityTag.class);
         PropertyParser.registerProperty(EntityTrades.class, EntityTag.class);
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
-            PropertyParser.registerProperty(EntityTranslation.class, EntityTag.class);
-        }
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_18)) {
-            PropertyParser.registerProperty(EntityTrapped.class, EntityTag.class);
-            PropertyParser.registerProperty(EntityTrapTime.class, EntityTag.class);
-        }
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_20)) {
-            PropertyParser.registerProperty(EntityVariant.class, EntityTag.class);
-        }
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
-            PropertyParser.registerProperty(EntityViewRange.class, EntityTag.class);
-        }
+        PropertyParser.registerProperty(EntityTranslation.class, EntityTag.class);
+        PropertyParser.registerProperty(EntityTrapped.class, EntityTag.class);
+        PropertyParser.registerProperty(EntityTrapTime.class, EntityTag.class);
+        PropertyParser.registerProperty(EntityVariant.class, EntityTag.class);
+        PropertyParser.registerProperty(EntityViewRange.class, EntityTag.class);
         PropertyParser.registerProperty(EntityVillagerExperience.class, EntityTag.class);
         PropertyParser.registerProperty(EntityVillagerLevel.class, EntityTag.class);
         PropertyParser.registerProperty(EntityVisible.class, EntityTag.class);
         PropertyParser.registerProperty(EntityVisualFire.class, EntityTag.class);
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
-            PropertyParser.registerProperty(EntityWidth.class, EntityTag.class);
-        }
+        PropertyParser.registerProperty(EntityWidth.class, EntityTag.class);
+
 
         // register core InventoryTag properties
         PropertyParser.registerProperty(InventoryContents.class, InventoryTag.class);
@@ -232,10 +_,8 @@
         registerItemProperty(ItemCanDestroy.class, "can_break");
         PropertyParser.registerProperty(ItemCanPlaceOn.class, ItemTag.class); // Let "can_place_on" through, this doesn't cover the entire component
         registerItemProperty(ItemColor.class, "dyed_color", "map_color"); // Potion color included in ItemPotion's "potion_contents"
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_20)) {
-            PropertyParser.registerProperty(ItemComponentsPatch.class, ItemTag.class);
-            registerItemProperty(ItemCustomData.class, "custom_data");
-        }
+        PropertyParser.registerProperty(ItemComponentsPatch.class, ItemTag.class);
+        registerItemProperty(ItemCustomData.class, "custom_data");
         registerItemProperty(ItemCustomModel.class, "custom_model_data");
         registerItemProperty(ItemChargedProjectile.class, "charged_projectiles");
         registerItemProperty(ItemEnchantments.class, "enchantments", "stored_enchantments");
@@ -244,9 +_,7 @@
         PropertyParser.registerProperty(ItemFlags.class, ItemTag.class);
         PropertyParser.registerProperty(ItemFrameInvisible.class, ItemTag.class); // Special case handling in ItemComponentsPatch
         PropertyParser.registerProperty(ItemHidden.class, ItemTag.class); // Relevant components control their own hiding internally
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
-            PropertyParser.registerProperty(ItemInstrument.class, ItemTag.class); // Special case handling in ItemComponentsPatch
-        }
+        PropertyParser.registerProperty(ItemInstrument.class, ItemTag.class); // Special case handling in ItemComponentsPatch
         registerItemProperty(ItemInventoryContents.class, "container", "bundle_contents");
         registerItemProperty(ItemKnowledgeBookRecipes.class, "recipes");
         registerItemProperty(ItemLock.class, "lock");
@@ -262,9 +_,7 @@
         registerItemProperty(ItemRepairCost.class, "repair_cost");
         PropertyParser.registerProperty(ItemScript.class, ItemTag.class);
         PropertyParser.registerProperty(ItemSignContents.class, ItemTag.class); // Special case handling in ItemComponentsPatch
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_20)) {
-            PropertyParser.registerProperty(ItemSignIsWaxed.class, ItemTag.class); // Special case handling in ItemComponentsPatch
-        }
+        PropertyParser.registerProperty(ItemSignIsWaxed.class, ItemTag.class); // Special case handling in ItemComponentsPatch
         registerItemProperty(ItemSkullskin.class, "profile");
         PropertyParser.registerProperty(ItemSpawnerCount.class, ItemTag.class); // Special case handling in ItemComponentsPatch
         PropertyParser.registerProperty(ItemSpawnerDelay.class, ItemTag.class); // Special case handling in ItemComponentsPatch
@@ -272,9 +_,7 @@
         PropertyParser.registerProperty(ItemSpawnerPlayerRange.class, ItemTag.class); // Special case handling in ItemComponentsPatch
         PropertyParser.registerProperty(ItemSpawnerRange.class, ItemTag.class); // Special case handling in ItemComponentsPatch
         PropertyParser.registerProperty(ItemSpawnerType.class, ItemTag.class); // Special case handling in ItemComponentsPatch
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_20)) {
-            registerItemProperty(ItemTrim.class, "trim");
-        }
+        registerItemProperty(ItemTrim.class, "trim");
         registerItemProperty(ItemUnbreakable.class, "unbreakable");
 
         // register core MaterialTag properties
@@ -299,9 +_,7 @@
         PropertyParser.registerProperty(MaterialLightable.class, MaterialTag.class);
         PropertyParser.registerProperty(MaterialMode.class, MaterialTag.class);
         PropertyParser.registerProperty(MaterialNote.class, MaterialTag.class);
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_21)) {
-            PropertyParser.registerProperty(MaterialOminous.class, MaterialTag.class);
-        }
+        PropertyParser.registerProperty(MaterialOminous.class, MaterialTag.class);
         PropertyParser.registerProperty(MaterialPersistent.class, MaterialTag.class);
         PropertyParser.registerProperty(MaterialPower.class, MaterialTag.class);
         PropertyParser.registerProperty(MaterialShape.class, MaterialTag.class);
@@ -312,17 +_,13 @@
         PropertyParser.registerProperty(MaterialWaterlogged.class, MaterialTag.class);
 
         // register core TradeTag properties
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_18)) {
-            PropertyParser.registerProperty(TradeDemand.class, TradeTag.class);
-        }
+        PropertyParser.registerProperty(TradeDemand.class, TradeTag.class);
         PropertyParser.registerProperty(TradeHasXp.class, TradeTag.class);
         PropertyParser.registerProperty(TradeInputs.class, TradeTag.class);
         PropertyParser.registerProperty(TradeMaxUses.class, TradeTag.class);
         PropertyParser.registerProperty(TradePriceMultiplier.class, TradeTag.class);
         PropertyParser.registerProperty(TradeResult.class, TradeTag.class);
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_18)) {
-            PropertyParser.registerProperty(TradeSpecialPrice.class, TradeTag.class);
-        }
+        PropertyParser.registerProperty(TradeSpecialPrice.class, TradeTag.class);
         PropertyParser.registerProperty(TradeUses.class, TradeTag.class);
         PropertyParser.registerProperty(TradeVillagerXP.class, TradeTag.class);
     }
