--- a/com/denizenscript/denizen/nms/v1_21/helpers/AdvancementHelperImpl.java
+++ b/com/denizenscript/denizen/nms/v1_21/helpers/AdvancementHelperImpl.java
@@ -2,9 +_,8 @@
 
 import com.denizenscript.denizen.nms.interfaces.AdvancementHelper;
 import com.denizenscript.denizen.nms.v1_21.Handler;
-import com.denizenscript.denizen.utilities.FormattedTextHelper;
+import com.denizenscript.denizen.utilities.PaperAPITools;
 import com.google.common.collect.ImmutableMap;
-import net.md_5.bungee.api.ChatColor;
 import net.minecraft.advancements.*;
 import net.minecraft.advancements.criterion.ImpossibleTrigger;
 import net.minecraft.core.ClientAsset;
@@ -175,7 +_,7 @@
                 ? getNMSAdvancementManager().advancements.get(CraftNamespacedKey.toMinecraft(advancement.parent))
                 : null;
         DisplayInfo display = new DisplayInfo(CraftItemStack.asNMSCopy(advancement.icon),
-                Handler.componentToNMS(FormattedTextHelper.parse(advancement.title, ChatColor.WHITE)), Handler.componentToNMS(FormattedTextHelper.parse(advancement.description, ChatColor.WHITE)),
+                Handler.parseNMSComponent(advancement.title, PaperAPITools.BaseColor.WHITE), Handler.parseNMSComponent(advancement.description, PaperAPITools.BaseColor.WHITE),
                 Optional.ofNullable(advancement.background).map(CraftNamespacedKey::toMinecraft).map(ClientAsset.ResourceTexture::new), AdvancementType.valueOf(advancement.frame.name()),
                 advancement.toast, advancement.announceToChat, advancement.hidden);
         display.setLocation(advancement.xOffset, advancement.yOffset);
