--- a/com/denizenscript/denizen/nms/v1_21/helpers/PlayerHelperImpl.java
+++ b/com/denizenscript/denizen/nms/v1_21/helpers/PlayerHelperImpl.java
@@ -17,7 +_,7 @@
 import com.denizenscript.denizen.objects.ItemTag;
 import com.denizenscript.denizen.objects.LocationTag;
 import com.denizenscript.denizen.objects.PlayerTag;
-import com.denizenscript.denizen.utilities.FormattedTextHelper;
+import com.denizenscript.denizen.utilities.PaperAPITools;
 import com.denizenscript.denizen.utilities.entity.DenizenEntityType;
 import com.denizenscript.denizen.utilities.entity.FakeEntity;
 import com.denizenscript.denizencore.objects.Mechanism;
@@ -25,7 +_,6 @@
 import com.denizenscript.denizencore.utilities.debugging.Debug;
 import com.mojang.authlib.GameProfile;
 import it.unimi.dsi.fastutil.ints.IntList;
-import net.md_5.bungee.api.ChatColor;
 import net.minecraft.core.Registry;
 import net.minecraft.core.registries.BuiltInRegistries;
 import net.minecraft.network.protocol.Packet;
@@ -400,7 +_,7 @@
 
     @Override
     public void setBossBarTitle(BossBar bar, String title) {
-        ((CraftBossBar) bar).getHandle().name = Handler.componentToNMS(FormattedTextHelper.parse(title, ChatColor.WHITE));
+        ((CraftBossBar) bar).getHandle().name = Handler.parseNMSComponent(title, PaperAPITools.BaseColor.WHITE);
         ((CraftBossBar) bar).getHandle().broadcast(ClientboundBossEventPacket::createUpdateNamePacket);
     }
 
@@ -453,7 +_,7 @@
         }
         GameProfile profile = ProfileEditorImpl.createGameProfile(id, name, texture, signature);
         // TODO: 1.21.3: Player list order and hat visibility support
-        ClientboundPlayerInfoUpdatePacket.Entry entry = new ClientboundPlayerInfoUpdatePacket.Entry(id, profile, listed, latency, gameMode == null ? null : GameType.byId(gameMode.getValue()), display == null ? null : Handler.componentToNMS(FormattedTextHelper.parse(display, ChatColor.WHITE)), true, player.getPlayerListOrder(), null);
+        ClientboundPlayerInfoUpdatePacket.Entry entry = new ClientboundPlayerInfoUpdatePacket.Entry(id, profile, listed, latency, gameMode == null ? null : GameType.byId(gameMode.getValue()), display == null ? null : Handler.parseNMSComponent(display, PaperAPITools.BaseColor.WHITE), true, player.getPlayerListOrder(), null);
         PacketHelperImpl.send(player, ProfileEditorImpl.createInfoPacket(actions, List.of(entry)));
     }
 
@@ -481,7 +_,7 @@
         nmsPlayer.connection.send(new ClientboundRespawnPacket(nmsPlayer.createCommonSpawnInfo(nmsWorld), ClientboundRespawnPacket.KEEP_ALL_DATA));
         nmsPlayer.connection.internalTeleport(PositionMoveRotation.of(nmsPlayer), Set.of());
         if (nmsPlayer.isPassenger()) {
-           nmsPlayer.connection.send(new ClientboundSetPassengersPacket(nmsPlayer.getVehicle()));
+            nmsPlayer.connection.send(new ClientboundSetPassengersPacket(nmsPlayer.getVehicle()));
         }
         if (nmsPlayer.isVehicle()) {
             nmsPlayer.connection.send(new ClientboundSetPassengersPacket(nmsPlayer));
