--- a/com/denizenscript/denizen/utilities/command/ExSustainedCommandHandler.java
+++ b/com/denizenscript/denizen/utilities/command/ExSustainedCommandHandler.java
@@ -3,7 +_,7 @@
 import com.denizenscript.denizen.Denizen;
 import com.denizenscript.denizen.objects.NPCTag;
 import com.denizenscript.denizen.objects.PlayerTag;
-import com.denizenscript.denizen.utilities.FormattedTextHelper;
+import com.denizenscript.denizen.utilities.PaperAPITools;
 import com.denizenscript.denizen.utilities.Settings;
 import com.denizenscript.denizen.utilities.depends.Depends;
 import com.denizenscript.denizen.utilities.implementation.BukkitScriptEntryData;
@@ -72,7 +_,7 @@
         }
         TimedQueue queue = getOrMakeQueue(sender instanceof Player player ? player : null, quiet);
         if (!quiet && sender instanceof Player) {
-            queue.debugOutput = s -> sender.spigot().sendMessage(FormattedTextHelper.parse(s.replace("<FORCE_ALIGN>", ""), net.md_5.bungee.api.ChatColor.WHITE));
+            queue.debugOutput = s -> PaperAPITools.instance.sendMessage(sender, s.replace("<FORCE_ALIGN>", ""));
         }
         else {
             queue.debugOutput = null;
