--- a/com/denizenscript/denizen/events/ScriptEventRegistry.java
+++ b/com/denizenscript/denizen/events/ScriptEventRegistry.java
@@ -80,35 +_,25 @@
         ScriptEvent.registerScriptEvent(BlockShearEntityScriptEvent.class);
         ScriptEvent.registerScriptEvent(BlockSpreadsScriptEvent.class);
         ScriptEvent.registerScriptEvent(BrewingStandFueledScriptEvent.class);
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
-            ScriptEvent.registerScriptEvent(BrewingStartsScriptEvent.class);
-        }
         ScriptEvent.registerScriptEvent(BrewsScriptEvent.class);
         ScriptEvent.registerScriptEvent(CauldronLevelChangeScriptEvent.class);
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_21)) {
-            ScriptEvent.registerScriptEvent(CrafterCraftsScriptEvent.class);
-        }
         ScriptEvent.registerScriptEvent(DragonEggMovesScriptEvent.class);
         ScriptEvent.registerScriptEvent(FurnaceBurnsItemScriptEvent.class);
         ScriptEvent.registerScriptEvent(FurnaceStartsSmeltingScriptEvent.class);
         ScriptEvent.registerScriptEvent(LeafDecaysScriptEvent.class);
         ScriptEvent.registerScriptEvent(LiquidLevelChangeScriptEvent.class);
         ScriptEvent.registerScriptEvent(LiquidSpreadScriptEvent.class);
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_21)) {
-            ScriptEvent.registerScriptEvent(LootDispensesFromBlockScriptEvent.class);
-        }
+        ScriptEvent.registerScriptEvent(LootDispensesFromBlockScriptEvent.class);
         ScriptEvent.registerScriptEvent(MoistureChangeScriptEvent.class);
         ScriptEvent.registerScriptEvent(NoteBlockPlaysNoteScriptEvent.class);
         ScriptEvent.registerScriptEvent(PistonExtendsScriptEvent.class);
         ScriptEvent.registerScriptEvent(PistonRetractsScriptEvent.class);
         ScriptEvent.registerScriptEvent(RedstoneScriptEvent.class);
         ScriptEvent.registerScriptEvent(SpongeAbsorbsScriptEvent.class);
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
-            ScriptEvent.registerScriptEvent(TNTPrimesScriptEvent.class);
-        }
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_21)) {
-            ScriptEvent.registerScriptEvent(VaultDisplaysItemScriptEvent.class);
-        }
+        ScriptEvent.registerScriptEvent(BrewingStartsScriptEvent.class);
+        ScriptEvent.registerScriptEvent(TNTPrimesScriptEvent.class);
+        ScriptEvent.registerScriptEvent(CrafterCraftsScriptEvent.class);
+        ScriptEvent.registerScriptEvent(VaultDisplaysItemScriptEvent.class);
 
         // Entity events
         ScriptEvent.registerScriptEvent(AreaEffectCloudApplyScriptEvent.class);
