--- a/com/denizenscript/denizen/nms/interfaces/ItemHelper.java
+++ b/com/denizenscript/denizen/nms/interfaces/ItemHelper.java
@@ -6,7 +_,6 @@
 import com.denizenscript.denizen.objects.ItemTag;
 import com.denizenscript.denizen.utilities.nbt.CustomNBT;
 import com.denizenscript.denizencore.objects.core.MapTag;
-import com.google.gson.JsonObject;
 import net.kyori.adventure.nbt.BinaryTag;
 import net.kyori.adventure.nbt.CompoundBinaryTag;
 import org.bukkit.DyeColor;
@@ -44,15 +_,7 @@
 
     public abstract String getJsonString(ItemStack itemStack);
 
-    public String getLegacyHoverNbt(ItemTag item) { // TODO: once 1.20 is the minimum supported version, remove this
-        return item.getItemMeta().getAsString();
-    }
-
-    public JsonObject getRawHoverComponentsJson(ItemStack item) {
-        throw new UnsupportedOperationException();
-    }
-
-    public ItemStack applyRawHoverComponentsJson(ItemStack item, JsonObject components) {
+    public ItemStack createItemWithNMSComponents(Material type, int count, Object nmsPatch) {
         throw new UnsupportedOperationException();
     }
 
@@ -85,7 +_,8 @@
     public ItemStack setEntityData(ItemStack item, CompoundBinaryTag entityNbt, EntityType entityType) { // TODO: once 1.20 is the minimum supported version, remove default impl
         boolean shouldRemove = entityNbt == null || entityNbt.isEmpty();
         CompoundBinaryTag nbt = getNbtData(item);
-        if (shouldRemove && !nbt.contains("EntityTag")) {
+        // TODO: adventure-nbt: contains
+        if (shouldRemove && !nbt.keySet().contains("EntityTag")) {
             return item;
         }
         if (shouldRemove) {
