--- a/com/denizenscript/denizen/objects/BiomeTag.java
+++ b/com/denizenscript/denizen/objects/BiomeTag.java
@@ -179,9 +_,7 @@
         // - narrate "The downfall type in plains biomes is: <biome[plains].downfall_type>!"
         // -->
         tagProcessor.registerTag(ElementTag.class, "downfall_type", (attribute, object) -> {
-            if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
-                BukkitImplDeprecations.biomeGlobalDownfallType.warn(attribute.context);
-            }
+            BukkitImplDeprecations.biomeGlobalDownfallType.warn(attribute.context);
             return new ElementTag(object.biome.getDownfallType());
         });
 
@@ -508,7 +_,7 @@
         });
     }
 
-    public static ObjectTagProcessor<BiomeTag> tagProcessor = new ObjectTagProcessor<>();
+    public static final ObjectTagProcessor<BiomeTag> tagProcessor = new ObjectTagProcessor<>();
 
     @Override
     public ObjectTag getObjectAttribute(Attribute attribute) {
