--- a/com/denizenscript/denizen/objects/EntityTag.java
+++ b/com/denizenscript/denizen/objects/EntityTag.java
@@ -1260,9 +_,6 @@
         // Deprecated in favor of <@link tag EntityTag.type> on MC 1.20+, which returns entity type names as specified by Mojang (scripts using this may need an update when switching).
         // -->
         tagProcessor.registerTag(ElementTag.class, "entity_type", (attribute, object) -> {
-            if (NMSHandler.getVersion().isAtMost(NMSVersion.v1_19)) {
-                return new ElementTag(object.getEntityType().getName(), true);
-            }
             BukkitImplDeprecations.oldSpigotNames.warn(attribute.context);
             return new ElementTag(switch (object.getEntityType().getName()) {
                 case "ITEM" -> "DROPPED_ITEM";
@@ -1370,22 +_,6 @@
                 return null;
             }
             return ScriptTag.valueOf(object.entityScript, CoreUtilities.noDebugContext);
-        });
-
-        // <--[tag]
-        // @attribute <EntityTag.scriptname>
-        // @returns ElementTag
-        // @deprecated use ".script.name" instead.
-        // @group data
-        // @description
-        // Use ".script.name" instead.
-        // -->
-        tagProcessor.registerTag(ElementTag.class, "scriptname", (attribute, object) -> {
-            BukkitImplDeprecations.hasScriptTags.warn(attribute.context);
-            if (object.entityScript == null) {
-                return null;
-            }
-            return new ElementTag(object.entityScript);
         });
 
         /////////////////////
