--- a/com/denizenscript/denizen/nms/v26_1/impl/network/handlers/packet/DenizenPacketHandlerPacketHandlers.java
+++ b/com/denizenscript/denizen/nms/v26_1/impl/network/handlers/packet/DenizenPacketHandlerPacketHandlers.java
@@ -8,6 +_,7 @@
 import net.minecraft.network.protocol.game.ClientGamePacketListener;
 import net.minecraft.network.protocol.game.ClientboundPlayerChatPacket;
 import net.minecraft.network.protocol.game.ClientboundSystemChatPacket;
+import org.bukkit.craftbukkit.util.CraftChatMessage;
 
 public class DenizenPacketHandlerPacketHandlers {
 
@@ -37,7 +_,7 @@
                         return null;
                     }
                     if (result.modified) {
-                        return new ClientboundSystemChatPacket(result.altMessageDetermination, isActionbar);
+                        return new ClientboundSystemChatPacket(CraftChatMessage.fromJSON(result.rawJson.asString()), isActionbar);
                     }
                 }
             }
