--- a/com/denizenscript/denizen/scripts/containers/core/BookScriptContainer.java
+++ b/com/denizenscript/denizen/scripts/containers/core/BookScriptContainer.java
@@ -2,13 +_,12 @@
 
 import com.denizenscript.denizen.objects.ItemTag;
 import com.denizenscript.denizen.tags.BukkitTagContext;
-import com.denizenscript.denizen.utilities.FormattedTextHelper;
+import com.denizenscript.denizen.utilities.PaperAPITools;
 import com.denizenscript.denizencore.objects.core.ScriptTag;
 import com.denizenscript.denizencore.scripts.containers.ScriptContainer;
 import com.denizenscript.denizencore.tags.TagContext;
 import com.denizenscript.denizencore.tags.TagManager;
 import com.denizenscript.denizencore.utilities.YamlConfiguration;
-import net.md_5.bungee.api.ChatColor;
 import org.bukkit.Material;
 import org.bukkit.inventory.meta.BookMeta;
 
@@ -94,8 +_,7 @@
         if (contains("text", List.class)) {
             List<String> pages = getStringList("text");
             for (String page : pages) {
-                page = TagManager.tag(page, context);
-                bookInfo.spigot().addPage(FormattedTextHelper.parse(page, ChatColor.BLACK));
+                PaperAPITools.instance.addPage(bookInfo, TagManager.tag(page, context));
             }
         }
         book.setItemMeta(bookInfo);
