--- a/com/denizenscript/denizen/scripts/containers/core/EnchantmentScriptContainer.java
+++ b/com/denizenscript/denizen/scripts/containers/core/EnchantmentScriptContainer.java
@@ -4,8 +_,7 @@
 import com.denizenscript.denizen.objects.EntityTag;
 import com.denizenscript.denizen.objects.ItemTag;
 import com.denizenscript.denizen.tags.BukkitTagContext;
-import com.denizenscript.denizen.utilities.FormattedTextHelper;
-import com.denizenscript.denizencore.utilities.debugging.Debug;
+import com.denizenscript.denizen.utilities.PaperAPITools;
 import com.denizenscript.denizen.utilities.implementation.BukkitScriptEntryData;
 import com.denizenscript.denizencore.objects.core.ElementTag;
 import com.denizenscript.denizencore.objects.core.ScriptTag;
@@ -18,8 +_,7 @@
 import com.denizenscript.denizencore.utilities.AsciiMatcher;
 import com.denizenscript.denizencore.utilities.CoreUtilities;
 import com.denizenscript.denizencore.utilities.YamlConfiguration;
-import net.md_5.bungee.api.ChatColor;
-import net.md_5.bungee.api.chat.BaseComponent;
+import com.denizenscript.denizencore.utilities.debugging.Debug;
 import org.bukkit.Bukkit;
 import org.bukkit.enchantments.Enchantment;
 import org.bukkit.entity.Entity;
@@ -225,7 +_,7 @@
 
     public List<String> slots;
 
-    public HashMap<Integer, BaseComponent[]> fullNamePerLevel = new HashMap<>();
+    public HashMap<Integer, String> fullNamePerLevel = new HashMap<>();
 
     public Enchantment enchantment;
 
@@ -280,13 +_,13 @@
         return CoreUtilities.toLowerCase(res).equals("true");
     }
 
-    public BaseComponent[] getFullName(int level) {
-        BaseComponent[] result = fullNamePerLevel.get(level);
+    public String getFullName(int level) {
+        String result = fullNamePerLevel.get(level);
         if (result != null) {
             return result;
         }
         String tagged = autoTagForLevel(fullNameTaggable, level);
-        result = FormattedTextHelper.parse(tagged, ChatColor.GRAY);
+        result = PaperAPITools.instance.parseTextToJson(tagged, PaperAPITools.BaseColor.GRAY);
         fullNamePerLevel.put(level, result);
         return result;
     }
