--- a/com/denizenscript/denizen/scripts/containers/core/ItemScriptHelper.java
+++ b/com/denizenscript/denizen/scripts/containers/core/ItemScriptHelper.java
@@ -62,15 +_,10 @@
         while (recipeIterator.hasNext()) {
             try {
                 if (recipeIterator.next() instanceof Keyed keyed && keyed.getKey().getNamespace().equals("denizen")) {
-                    if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_21)) {
-                        keys.add(keyed.getKey());
-                    }
-                    else {
-                        recipeIterator.remove();
-                    }
+                    keys.add(keyed.getKey());
                 }
             }
-            catch (AbstractMethodError ignored) {} // TODO: 26.1: work around Spigot bug
+            catch (AbstractMethodError ignored) {}
         }
         if (!keys.isEmpty()) {
             NMSHandler.itemHelper.removeRecipes(keys);
