--- a/com/denizenscript/denizen/paper/properties/PaperEntityExtensions.java
+++ b/com/denizenscript/denizen/paper/properties/PaperEntityExtensions.java
@@ -52,6 +_,23 @@
             return new ElementTag(experienceOrb.getSpawnReason());
         });
 
+        // <--[mechanism]
+        // @object EntityTag
+        // @name restock_trades
+        // @input None
+        // @Plugin Paper
+        // @group paper
+        // @description
+        // Restocks a villager's trades.
+        // Note: this mechanism will fire the <@link event villager replenishes trade> event for every trade the villager is offering.
+        // This mechanism also updates the villager's demand for offers, which may cause item trade prices to rise or fall.
+        // -->
+        EntityTag.registerSpawnedOnlyMechanism("restock_trades", false, (object, mechanism) -> {
+            if (object.getBukkitEntity() instanceof Villager villager) {
+                villager.restock();
+            }
+        });
+
         // <--[tag]
         // @attribute <EntityTag.xp_trigger>
         // @returns EntityTag
@@ -187,26 +_,6 @@
                 }
                 object.getLivingEntity().damageItemStack(slot.asEnum(EquipmentSlot.class), amount.asInt());
             });
-
-            if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_21)) {
-
-                // <--[mechanism]
-                // @object EntityTag
-                // @name restock_trades
-                // @input None
-                // @Plugin Paper
-                // @group paper
-                // @description
-                // Restocks a villager's trades.
-                // Note: this mechanism will fire the <@link event villager replenishes trade> event for every trade the villager is offering.
-                // This mechanism also updates the villager's demand for offers, which may cause item trade prices to rise or fall.
-                // -->
-                EntityTag.registerSpawnedOnlyMechanism("restock_trades", false, (object, mechanism) -> {
-                    if (object.getBukkitEntity() instanceof Villager villager) {
-                        villager.restock();
-                    }
-                });
-            }
 
             // <--[mechanism]
             // @object EntityTag
