--- a/com/denizenscript/denizen/paper/events/PlayerSetSpawnScriptEvent.java
+++ b/com/denizenscript/denizen/paper/events/PlayerSetSpawnScriptEvent.java
@@ -1,14 +_,14 @@
 package com.denizenscript.denizen.paper.events;
 
+import com.denizen.m.utilities.FormattedTextHelper;
 import com.denizenscript.denizen.events.BukkitScriptEvent;
 import com.denizenscript.denizen.objects.LocationTag;
-import com.denizenscript.denizen.paper.PaperModule;
 import com.denizenscript.denizen.utilities.implementation.BukkitScriptEntryData;
 import com.denizenscript.denizencore.objects.ObjectTag;
 import com.denizenscript.denizencore.objects.core.ElementTag;
 import com.denizenscript.denizencore.scripts.ScriptEntryData;
 import com.destroystokyo.paper.event.player.PlayerSetSpawnEvent;
-import net.md_5.bungee.api.ChatColor;
+import net.kyori.adventure.text.format.NamedTextColor;
 import org.bukkit.event.EventHandler;
 import org.bukkit.event.Listener;
 
@@ -58,7 +_,7 @@
             return false;
         });
         this.<PlayerSetSpawnScriptEvent, ElementTag>registerDetermination("message", ElementTag.class, (evt, context, message) -> {
-            evt.event.setNotification(PaperModule.parseFormattedText(message.toString(), ChatColor.WHITE));
+            evt.event.setNotification(FormattedTextHelper.parse(message.asString(), NamedTextColor.WHITE));
         });
         this.<PlayerSetSpawnScriptEvent, ElementTag>registerOptionalDetermination("notify", ElementTag.class, (evt, context, value) -> {
             if (value.isBoolean()) {
@@ -97,7 +_,7 @@
             case "cause" -> new ElementTag(event.getCause());
             case "forced" -> new ElementTag(event.isForced());
             case "location" -> event.getLocation() != null ? new LocationTag(event.getLocation()) : null;
-            case "message" -> event.getNotification() != null ? new ElementTag(PaperModule.stringifyComponent(event.getNotification()), true) : null;
+            case "message" -> event.getNotification() != null ? new ElementTag(FormattedTextHelper.stringify(event.getNotification()), true) : null;
             case "notify" -> new ElementTag(event.willNotifyPlayer());
             default -> super.getContext(name);
         };
