--- a/com/denizenscript/denizen/paper/events/PlayerItemTakesDamageScriptEventPaperImpl.java
+++ b/com/denizenscript/denizen/paper/events/PlayerItemTakesDamageScriptEventPaperImpl.java
@@ -1,8 +_,6 @@
 package com.denizenscript.denizen.paper.events;
 
 import com.denizenscript.denizen.events.player.PlayerItemTakesDamageScriptEvent;
-import com.denizenscript.denizen.nms.NMSHandler;
-import com.denizenscript.denizen.nms.NMSVersion;
 import com.denizenscript.denizencore.objects.ObjectTag;
 import com.denizenscript.denizencore.objects.core.ElementTag;
 
@@ -11,7 +_,7 @@
     @Override
     public ObjectTag getContext(String name) {
         switch (name) {
-            case "original_damage": return NMSHandler.getVersion().isAtLeast(NMSVersion.v1_18) ? new ElementTag(event.getOriginalDamage()) : null;
+            case "original_damage": return new ElementTag(event.getOriginalDamage());
         }
         return super.getContext(name);
     }
