--- a/com/denizenscript/denizen/paper/events/ProjectileCollideScriptEvent.java
+++ b/com/denizenscript/denizen/paper/events/ProjectileCollideScriptEvent.java
@@ -1,8 +_,6 @@
 package com.denizenscript.denizen.paper.events;
 
 import com.denizenscript.denizen.events.BukkitScriptEvent;
-import com.denizenscript.denizen.nms.NMSHandler;
-import com.denizenscript.denizen.nms.NMSVersion;
 import com.denizenscript.denizen.objects.EntityTag;
 import com.denizenscript.denizen.utilities.BukkitImplDeprecations;
 import com.denizenscript.denizen.utilities.implementation.BukkitScriptEntryData;
@@ -48,9 +_,8 @@
         if (!super.couldMatch(path)) {
             return false;
         }
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
-            BukkitImplDeprecations.projectileCollideEvent.warn(path.container);
-        }
+
+        BukkitImplDeprecations.projectileCollideEvent.warn(path.container);
         return true;
     }
 
