--- a/com/denizenscript/denizen/utilities/packets/DenizenPacketHandler.java
+++ b/com/denizenscript/denizen/utilities/packets/DenizenPacketHandler.java
@@ -69,7 +_,7 @@
 
     public static boolean isHoldingRaisable(Player player) {
         return PlayerRaiseLowerItemScriptEvent.raisableItems.contains(player.getEquipment().getItemInMainHand().getType())
-            || PlayerRaiseLowerItemScriptEvent.raisableItems.contains(player.getEquipment().getItemInOffHand().getType());
+                || PlayerRaiseLowerItemScriptEvent.raisableItems.contains(player.getEquipment().getItemInOffHand().getType());
     }
 
     public void receivePlacePacket(final Player player) {
@@ -113,8 +_,8 @@
         if (event.loaded) {
             Callable<PlayerReceivesMessageScriptEvent> eventCall = () -> {
                 event.reset();
-                event.message = new ElementTag(chat.getMessage());
-                event.rawJson = new ElementTag(chat.getRawJson());
+                event.message = new ElementTag(chat.getMessage(), true);
+                event.rawJson = new ElementTag(chat.getRawJson(), true);
                 event.system = new ElementTag(chat.isSystem());
                 event.player = PlayerTag.mirrorBukkitPlayer(player);
                 return event.triggerNow();
