--- a/com/denizenscript/denizen/scripts/commands/player/ActionBarCommand.java
+++ b/com/denizenscript/denizen/scripts/commands/player/ActionBarCommand.java
@@ -2,7 +_,7 @@
 
 import com.denizenscript.denizen.objects.PlayerTag;
 import com.denizenscript.denizen.tags.BukkitTagContext;
-import com.denizenscript.denizen.utilities.FormattedTextHelper;
+import com.denizenscript.denizen.utilities.PaperAPITools;
 import com.denizenscript.denizen.utilities.Utilities;
 import com.denizenscript.denizencore.exceptions.InvalidArgumentsException;
 import com.denizenscript.denizencore.objects.Argument;
@@ -16,8 +_,6 @@
 import com.denizenscript.denizencore.scripts.containers.core.FormatScriptContainer;
 import com.denizenscript.denizencore.tags.TagManager;
 import com.denizenscript.denizencore.utilities.debugging.Debug;
-import net.md_5.bungee.api.ChatColor;
-import net.md_5.bungee.api.ChatMessageType;
 
 import java.util.Collections;
 import java.util.List;
@@ -135,7 +_,7 @@
                     context.player = player;
                     personalText = TagManager.tag(personalText, context);
                 }
-                player.getPlayerEntity().spigot().sendMessage(ChatMessageType.ACTION_BAR, FormattedTextHelper.parse(format != null ? format.getFormattedText(personalText, scriptEntry) : personalText, ChatColor.WHITE));
+                PaperAPITools.instance.sendActionBar(player.getPlayerEntity(), format != null ? format.getFormattedText(personalText, scriptEntry) : personalText);
             }
             else {
                 Debug.echoError("Sent actionbar to non-existent player!?");
