--- a/com/denizenscript/denizen/scripts/commands/player/ToastCommand.java
+++ b/com/denizenscript/denizen/scripts/commands/player/ToastCommand.java
@@ -113,7 +_,7 @@
         if (scriptEntry.dbCallShouldDebug()) {
             Debug.report(scriptEntry, name, text, frame, icon, db("targets", targets));
         }
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_20) && icon.getBukkitMaterial().isAir()) {
+        if (icon.getBukkitMaterial().isAir()) {
             icon = new ItemTag(Material.DIRT);
         }
         final Advancement advancement = new Advancement(true,
