--- a/com/denizenscript/denizen/utilities/debugging/DebugConsoleSender.java
+++ b/com/denizenscript/denizen/utilities/debugging/DebugConsoleSender.java
@@ -19,7 +_,7 @@
         //                                                                       "[HH:mm:ss INFO]: "
         string = CoreConfiguration.debugPrefix + string.replace("<FORCE_ALIGN>", "                 ");
         if (showColor) {
-            PaperAPITools.instance.sendConsoleMessage(commandSender, string);
+            PaperAPITools.instance.sendMessage(commandSender, string);
         }
         else {
             commandSender.sendMessage(ChatColor.stripColor(string));
