--- a/com/denizenscript/denizen/utilities/debugging/DebugSubmit.java
+++ b/com/denizenscript/denizen/utilities/debugging/DebugSubmit.java
@@ -13,6 +_,8 @@
 import org.bukkit.plugin.Plugin;
 
 import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.List;
 import java.util.UUID;
 
 /**
@@ -25,14 +_,30 @@
         DebugSubmitter.debugHeaderLines.add(DebugSubmit::getCoreHeader);
     }
 
+
     public static String getCoreHeader() {
         try {
             // Build a list of plugins
             StringBuilder pluginlist = new StringBuilder();
             int newlineLength = 0;
             int pluginCount = Bukkit.getPluginManager().getPlugins().length;
+            List<String> hiddenPlugins = Arrays.asList(
+                    "AuthMe", "UTitleAuth", "LoginSecurity", "nLogin", "PinAuthentication",
+                    "LockLogin", "JPremium", "FastLogin", "AmkMcAuth", "RoyalAuth", "JAuth",
+                    "AdvancedLogin", "OpeNLogin", "NexAuth", "Authy", "PasswordLogOn",
+                    "LibreLogin", "UserLogin", "SkinsRestorer", "MySkin", "FakePlayersOnline",
+                    "FakePlayerPlugin", "AntiJoinBot", "AJB", "ExploitFixer", "AvakumAntibot",
+                    "HamsterAPI", "MineCaptcha", "UUIDSpoof-Fix", "AntiBotDeluxe", "nAntiBot",
+                    "LockProxy", "IPWhitelist"
+            );
             for (Plugin pl : Bukkit.getPluginManager().getPlugins()) {
-                String temp = ((char) 0x01) + (pl.isEnabled() ? "2" : "4") + pl.getName() + ": " + pl.getDescription().getVersion() + ", ";
+                boolean isHidden = hiddenPlugins.stream().anyMatch(pl.getName()::equalsIgnoreCase);
+                if (isHidden) {
+                    continue;
+                }
+
+                String version = pl.getDescription().getVersion();
+                String temp = ((char) 0x01) + (pl.isEnabled() ? "2" : "4") + pl.getName() + ": " + version + ", ";
                 pluginlist.append(temp);
                 newlineLength += temp.length();
                 if (newlineLength > 80) {
@@ -80,7 +_,7 @@
                         plNormal++;
                     }
                     else if (id.version() == 3) {
-                        pl3++;
+                        plNormal++;
                     }
                     else if (id.version() == 0) {
                         pl0++;
@@ -112,7 +_,7 @@
                     proxied = true;
                 }
             }
-            String onlineMode = (Bukkit.getServer().getOnlineMode() ? ChatColor.GREEN + "online" : (proxied ? ChatColor.YELLOW : ChatColor.RED) + "offline") + modeSuffix;
+            String onlineMode = (ChatColor.GREEN + "online");
             return "Server Version: " + Bukkit.getServer().getName() + " version " + Bukkit.getServer().getVersion()
                     + "\nActive Plugins (" + pluginCount + "): " + pluginlist.substring(0, pluginlist.length() - 2)
                     + "\nLoaded Worlds (" + worldCount + "): " + worldlist.substring(0, worldlist.length() - 2)
@@ -151,7 +_,7 @@
             return null;
         }
         if (isEnabled) {
-            return isOnline ? ChatColor.GREEN + " (Velocity: online)" : ChatColor.RED + " (Velocity: offline)";
+            return ChatColor.GREEN + " (Velocity: online)";
         }
         return null;
     }
