--- a/com/denizenscript/denizen/scripts/commands/item/InventoryCommand.java
+++ b/com/denizenscript/denizen/scripts/commands/item/InventoryCommand.java
@@ -2,7 +_,6 @@
 
 import com.denizenscript.denizen.Denizen;
 import com.denizenscript.denizen.nms.NMSHandler;
-import com.denizenscript.denizen.nms.NMSVersion;
 import com.denizenscript.denizen.objects.EntityTag;
 import com.denizenscript.denizen.objects.InventoryTag;
 import com.denizenscript.denizen.objects.ItemTag;
@@ -223,7 +_,7 @@
         if (!event.getPlayer().getUniqueId().equals(currentAltPlayer.getUniqueId())) {
             return;
         }
-        if (currentAltTitle != null && NMSHandler.getVersion().isAtLeast(NMSVersion.v1_18)) {
+        if (currentAltTitle != null) {
             NMSHandler.getInstance().setInventoryTitle(event.getView(), currentAltTitle);
         }
         InventoryTag newTag = new InventoryTag(event.getInventory(), currentAltType, currentAltHolder);
@@ -444,6 +_,7 @@
                     Argument mechanismArgument = new Argument(dataAction);
                     boolean hasValue = mechanismArgument.hasPrefix();
                     toAdjust.safeAdjust(new Mechanism(hasValue ? mechanismArgument.getPrefix().getValue() : mechanismArgument.getValue(), hasValue ? mechanismArgument.object : null, scriptEntry.getContext()));
+                    toAdjust.safeAdjust(new Mechanism(mechanismArgument.getPrefix().getValue(), mechanismArgument.object, scriptEntry.getContext()));
                     NMSHandler.itemHelper.setInventoryItem(destination.getInventory(), toAdjust.getItemStack(), slotId);
                     break;
                 case FLAG:
