--- a/com/denizenscript/denizen/nms/v1_21/impl/SidebarImpl.java
+++ b/com/denizenscript/denizen/nms/v1_21/impl/SidebarImpl.java
@@ -3,9 +_,8 @@
 import com.denizenscript.denizen.nms.abstracts.Sidebar;
 import com.denizenscript.denizen.nms.v1_21.Handler;
 import com.denizenscript.denizen.nms.v1_21.helpers.PacketHelperImpl;
-import com.denizenscript.denizen.utilities.FormattedTextHelper;
+import com.denizenscript.denizen.utilities.PaperAPITools;
 import com.denizenscript.denizencore.utilities.debugging.Debug;
-import net.md_5.bungee.api.ChatColor;
 import net.minecraft.network.chat.Component;
 import net.minecraft.network.chat.numbers.StyledFormat;
 import net.minecraft.network.protocol.game.ClientboundSetDisplayObjectivePacket;
@@ -47,7 +_,7 @@
 
     public SidebarImpl(Player player) {
         super(player);
-        Component chatComponentTitle = Handler.componentToNMS(FormattedTextHelper.parse(title, ChatColor.WHITE));
+        Component chatComponentTitle = Handler.parseNMSComponent(title, PaperAPITools.BaseColor.WHITE);
         this.obj1 = new Objective(dummyScoreboard, "dummy_1", dummyCriteria, chatComponentTitle, ObjectiveCriteria.RenderType.INTEGER, false, StyledFormat.SIDEBAR_DEFAULT);
         this.obj2 = new Objective(dummyScoreboard, "dummy_2", dummyCriteria, chatComponentTitle, ObjectiveCriteria.RenderType.INTEGER, false, StyledFormat.SIDEBAR_DEFAULT);
     }
@@ -55,7 +_,7 @@
     @Override
     protected void setDisplayName(String title) {
         if (this.obj1 != null) {
-            Component chatComponentTitle = Handler.componentToNMS(FormattedTextHelper.parse(title, ChatColor.WHITE));
+            Component chatComponentTitle = Handler.parseNMSComponent(title, PaperAPITools.BaseColor.WHITE);
             this.obj1.setDisplayName(chatComponentTitle);
             this.obj2.setDisplayName(chatComponentTitle);
         }
@@ -77,7 +_,7 @@
             String lineId = ids[i];
             PlayerTeam team = new PlayerTeam(dummyScoreboard, lineId);
             team.getPlayers().add(lineId);
-            team.setPlayerPrefix(Handler.componentToNMS(FormattedTextHelper.parse(line, ChatColor.WHITE)));
+            team.setPlayerPrefix(Handler.parseNMSComponent(line, PaperAPITools.BaseColor.WHITE));
             generatedTeams.add(team);
             PacketHelperImpl.send(player, ClientboundSetPlayerTeamPacket.createAddOrModifyPacket(team, true));
             PacketHelperImpl.send(player, new ClientboundSetScorePacket(lineId, obj1.getName(), this.scores[i], Optional.empty(), Optional.of(StyledFormat.SIDEBAR_DEFAULT)));
